integrationtests/as-lucene-directory/src/test/java/org/infinispan/test/integration/as/wildfly/InfinispanModuleMemberRegistrationIT.java

```diff
@@ -128,10 +128,52 @@ public void testNewMemberSearch() throws Exception {
    @Test
    @InSequence(value = 3)
    @OperateOnDeployment("dep.active-2")
+   public void testNewMemberLuceneSearch() throws Exception {
+      List<Member> search = memberRegistration.luceneSearch("Peter");
+
+      assertFalse("Expected at least one result after the indexing", search.isEmpty());
+      assertEquals("Lucene search hasn't found a member", "Peter O'Tall", search.get(0).getName());
+   }
+
+   @Test
+   @InSequence(value = 4)
+   @OperateOnDeployment("dep.active-2")
    public void testNonExistingMember() throws Exception {
       List<Member> search = memberRegistration.search("TotallyInventedName");
 
       assertNotNull("Search should never return null", search);
       assertTrue("Search results should be empty", search.isEmpty());
    }
+
+   @Test
+   @InSequence(value = 5)
+   @OperateOnDeployment("dep.active-2")
+   public void testLuceneNonExistingMember() throws Exception {
+      List<Member> search = memberRegistration.luceneSearch("TotallyInventedName");
+
+      assertNotNull("Search should never return null", search);
+      assertTrue("Search results should be empty", search.isEmpty());
+   }
+
+   @Test
+   @InSequence(value = 6)
+   @OperateOnDeployment("dep.active-2")
+   public void testPurgeIndex() throws Exception {
+      memberRegistration.purgeMemberIndex();
+      List<Member> search = memberRegistration.search("Peter");
+
+      assertNotNull("Search should never return null", search);
+      assertTrue("Search results should be empty", search.isEmpty());
+   }
+
+   @Test
+   @InSequence(value = 7)
+   @OperateOnDeployment("dep.active-2")
+   public void testReIndex() throws Exception {
```

Comment:
So, I was confused as I thought this was an existing test (hence that it was supposed to work), but the test is wrong.

The problem is that method _testReIndex_ is invoking the _memberRegistration_ bean multiple times.
Essentialy the bug is in _org.infinispan.test.integration.as.wildfly.util.Resources_ : we can't wrap an EntityManager which is injected via a '@PersistenceContext' and use that to produce the CDI reference, as the 'EntityManager' which is being injected is getting closed on each invocation of the EJB.
At the second invocation of _memberRegistration_, the EntityManager is closed so the reference to the 'FullTextEntityManager' is invalid.

One solution is to never have two invocations to _memberRegistration_ (as in my original tests). Another possibly more elegant solution is to get rid of the flawed 'Resources' helper, and inject the EntityManager always using the '@PersistenceContext' annotation.